File: 339b/main.cpp
#include <stdlib.h>
#include <stdio.h>
#include <iostream>
using namespace std;

int main() {
    int n, m, v;
    scanf("%d %d", &n, &m);
    
    int prev = 1;
    long long step = 0;
    int i = 0;
    for (; i < m; i++) {
        scanf("%d", &v);
        if (v >= prev) {
            step += (v - prev);
        } else {
            step += (n + v - prev);
        }
        prev = v;
    }

    cout << step;
}

File: 339b/main.go
package main

import "fmt"

func main() {
	var n, m, v int
	fmt.Scan(&n, &m)

	step := 0
	prev := 1
	for i := 0; i < m; i++ {
		fmt.Scan(&v)
		if v >= prev {
			step += (v - prev)
		} else {
			step += (n + v - prev)
		}
		prev = v
	}

	fmt.Print(step)
}
